### PR TITLE
add draid test helper functions

### PR DESCRIPTION
**Pull request by richardelling**
Add draid test helper functions. Improved robustness for vdevs_in_pool

### Motivation and Context
The draid feature changes some of the responses to zpool status 
that are helpful for writing ZTS tests. This adds a few helper functions
similar to the existing functions for other features.

Also, the previous vdevs_in_pool function can fail in ambiguous ways.
This is changed to be more precise and robust.

### Description
Added functions:

- wait_vdev_gone
- wait_pool_resilvered
- wait_pool_rebuilt
- is_pool_rebuilding
- is_pool_rebuilt

### How Has This Been Tested?
Manual testing of functions. Also tested in draid test scripts to be integrated with a
later PR.

### Types of changes

- [ x] Bug fix (non-breaking change which fixes an issue)
- [ x] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [x] My code follows the ZFS on Linux [code style requirements](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**contributing** document](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/zfsonlinux/zfs/tree/master/tests) to cover my changes.
- [ ] All new and existing tests passed.
- [ ] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).
- [ ] Change has been approved by a ZFS on Linux member.

### CODE REVIEW

**Comment by richardelling** [comment]
pausing to add shunit2 tests for new functions

File: tests/runfiles/linux.run [hunk 1]
```diff
@@ -873,3 +873,14 @@ tags = ['functional', 'zvol', 'zvol_swap']
 [tests/functional/libzfs]
 tests = ['many_fds', 'libzfs_input']
 tags = ['functional', 'libzfs']
+
+
+# Unit tests are separate from functional tests because they cover the
```
Comment (jwk404) [inline tests/runfiles/linux.run]: For these test groups, if you add `pre =` and `post =` then you can omit the setup and cleanup files, which aren't doing anything at the moment.

File: tests/zfs-tests/tests/unit/libtest/Makefile.am [hunk 2]
```diff
@@ -0,0 +1,5 @@
+pkgdatadir = $(datadir)/@PACKAGE@/zfs-tests/tests/unit/libtest
+dist_pkgdata_SCRIPTS = \
+	libtest_001_pos.ksh \
```
Comment (jwk404) [inline tests/zfs-tests/tests/unit/libtest/Makefile.am]: Could we rename these to have meaningful filenames describing what's tested?

File: tests/zfs-tests/tests/unit/math/Makefile.am [hunk 3]
```diff
@@ -0,0 +1,3 @@
+pkgdatadir = $(datadir)/@PACKAGE@/zfs-tests/tests/unit/math
+dist_pkgdata_SCRIPTS = \
+	math_001_pos.ksh
```
Comment (jwk404) [inline tests/zfs-tests/tests/unit/math/Makefile.am]: Rename here too?

File: tests/zfs-tests/include/libtest.shlib [hunk 4]
```diff
@@ -2527,10 +2603,18 @@ function get_compress_opts
 		COMPRESS_OPTS="on off lzjb"
 	fi
 	typeset valid_opts="$COMPRESS_OPTS"
-	zfs get 2>&1 | grep gzip >/dev/null 2>&1
+	zfs get 2>&1 | grep -w compression | grep gzip >/dev/null 2>&1
```
[the remark this answers is not in the document]
Comment (richardelling) [reply tests/zfs-tests/include/libtest.shlib]: I debated that myself. I'll make that change.

File: tests/runfiles/linux.run (hunk 1 above)
Comment (richardelling) [reply tests/runfiles/linux.run]: good idea, less is more

File: tests/zfs-tests/tests/unit/libtest/Makefile.am (hunk 2 above)
Comment (richardelling) [reply tests/zfs-tests/tests/unit/libtest/Makefile.am]: it is testing libtest.shlib, no?  or do you mean in finer detail, such as libtest_vdev_001_pos.ksh? or perhaps libtest.shlib_001_pos.ksh?

Comment (jwk404) [reply tests/zfs-tests/tests/unit/libtest/Makefile.am]: I'm thinking something more like check_pool_status.ksh vdev_tests.ksh and maybe utils_tests.ksh. Granted they all test libtest.shlib, but the contents of each file is a logical group of similar tests. With more descriptive names, when future authors want to know if there's a file they can add a unit test to as opposed to creating a new one descriptive names will be more helpful than libtest_0*.ksh.

Comment (richardelling) [reply tests/zfs-tests/tests/unit/libtest/Makefile.am]: good idea, I never liked the numbering anyway :-)

**Comment by richardelling** [comment]
closing this PR, will revisit later as part of the draid project

**Pull request closed without merging**